File: Article_6/cpp/classes.cpp
#include <iostream>
#include <string>

class Character
{
    public:
        std::string name;
        std::string race;
        std::string specialty;
        int hp;
        int mp;

        std::string setName()
        {
            std::cout << "What is your name?\n";
            std::cin >> name;
            return name;
        };

        std::string setRace()
        {
            std::cout << "What is your race? Choose between human, elf, or dwarf.\n";
            std::cin >> race;
            return race;
        };

        std::string setSpecialty()
        {
            std::cout << "What is your specialty (or class)? Choose between warrior, mage, or rogue.\n";
            std::cin >> specialty;
            return specialty;
        };

        int setHP()
        {

            if (specialty == "Warrior")
            {
                hp = 25;
            }
            else if (specialty == "Mage")
            {
                hp = 15;
            }
            else if (specialty == "Rogue")
            {
                hp = 20;
            }

            std::cout << "Your hp is " << hp << std::endl;
            return hp;
        };

        int setMP()
        {

            if (specialty == "Warrior")
            {
                mp = 5;
            }
            else if (specialty == "Mage")
            {
                mp = 15;
            }
            else if (specialty == "Rogue")
            {
                mp = 10;
            }

            std::cout << "Your mp is " << mp << std::endl;
            return mp;
        };

        void viewCharStats()
        {
            std::cout << "Name: " << name << std::endl;
            std::cout << "Race: " << race << std::endl;
            std::cout << "Class: " << specialty << std::endl;
            std::cout << "HP: " << hp << std::endl;
            std::cout << "MP: " << mp << std::endl;
        }


};

int main()
{
    Character myChar; // Create new character object
    myChar.setName(); // Call setName method
    myChar.setRace(); // Call setRace method
    myChar.setSpecialty(); // Call setSpeciality method
    myChar.setHP(); // Call setHP method
    myChar.setMP(); // Call setMP method
    myChar.viewCharStats(); // Call viewCharStats method

    return 0;
}

File: Article_6/python/classes.py
class Character():
    """ A class that holds character information """

    def __init__(self, name="", race="", specialty="", hp=0, mp=0):
        self.name = name
        self.race = race
        self.specialty = specialty
        self.hp = hp
        self.mp = mp

    def set_name(self):
        """ Sets character name """
        print("What is your name?")
        self.name = input()

    def set_race(self):
        """ Sets character race """
        print("What is your race? Choose between human, elf, or dwarf.")
        self.race = input()

    def set_specialty(self):
        """ Sets character class/specialty """
        print("What is your specialty (or class)? Choose between warrior, mage, or rogue.")
        self.specialty = input()

    def set_hp(self):
        if self.specialty == "Warrior":
            self.hp = 25
        elif self.specialty == "Mage":
            self.hp = 15
        elif self.specialty == "Rogue":
            self.hp = 20
        print(f"You have {self.hp} hit points.")

    def set_mp(self):
        if self.specialty == "Warrior":
            self.mp = 5
        elif self.specialty == "Mage":
            self.mp = 15
        elif self.specialty == "Rogue":
            self.mp = 10
        print(f"You have {self.mp} magic points.")

    def view_character_stats(self):
        print(f"Name: {self.name}")
        print(f"Race: {self.race}")
        print(f"Class: {self.specialty}")
        print(f"HP: {self.hp}")
        print(f"MP: {self.mp}")


newChar = Character()
newChar.set_name()
newChar.set_race()
newChar.set_specialty()
newChar.set_hp()
newChar.set_mp()
newChar.view_character_stats()